### PR TITLE
Mouse in hay minigame

### PR DESCRIPTION
**Pull request by Kiuh**

### CODE REVIEW

File: Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/ScoreCounter.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        [SerializeField]
+        private Transition.End.Controller endController;
+
+        [SerializeField]
+        private InputAction quitGameInputAction;
+
+        [Scene]
+        [SerializeField]
+        private string nextScene;
```
Comment (ic3mintt) [inline Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/ScoreCounter.cs]: It shouldn't be here

Comment (ic3mintt) [reply Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/ScoreCounter.cs]: Everything that is connected to Scene Change should be in separate class

File: Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/ScoreCounter.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            Debug.Log("Exit by key");
+            ExitGame();
+        }
+
+        public void AddPoint()
+        {
+            score++;
+        }
+
+        public bool IsWinGame => score == neededScore;
```
Comment (ic3mintt) [inline Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/ScoreCounter.cs]: Move up to fields

File: Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/ScoreCounter.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        private void Update()
+        {
+            scoreLabel.text = score.ToString();
```
Comment (ic3mintt) [inline Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/ScoreCounter.cs:24]: Do not use it in Update. Move it to AddPoint method

File: Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/Paw.cs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+using UnityEngine.Events;
+using UnityEngine.InputSystem;
+
+namespace Minigames.MouseInHay
+{
+    [AddComponentMenu("Scripts/Minigames/MouseInHay/Minigames.MouseInHay.Paw")]
+    internal class Paw : MonoBehaviour
+    {
+        [Serializable]
+        private struct HoleByInputAction
```
Comment (ic3mintt) [inline Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/Paw.cs:17]: move it out of the class, please

File: Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/Paw.cs [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+        private void Awake()
+        {
+            foreach (KeyValuePair<Hole, InputAction> pair in HolesWithInput)
+            {
+                pair.Value.performed += (context) => InputPerformed(pair.Key);
+            }
+        }
+
+        private void InputPerformed(Hole hole)
```
Comment (ic3mintt) [inline Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/Paw.cs]:
```suggestion
        private void MoveTo(Hole hole)
```

File: Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/Paw.cs [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+            isInAction = true;
+            TweenerCore<Vector3, Vector3, VectorOptions> moveToHole = transform.DOMove(
+                hole.transform.position,
+                moveToHoleTime
+            );
+            moveToHole.onComplete += () => GrabMouse(hole);
+        }
+
+        private void GrabMouse(Hole hole)
```
Comment (ic3mintt) [inline Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/Paw.cs]: TrySlapMouse or TryGrabMouse

Comment (Kiuh) [reply Fantasy-Quest/Assets/Scripts/Minigames/MouseInHay/Paw.cs]: Prefix `Try` implies return bool and do something or nothing, this method do many things.

**Review by ic3mintt — APPROVED** [review]

**Pull request merged**